include/bluetooth/services/fast_pair.h

```diff
@@ -30,18 +37,82 @@ enum bt_fast_pair_adv_mode {
 
 	/** Fast Pair not discoverable advertising, hide UI indication. */
 	BT_FAST_PAIR_ADV_MODE_NOT_DISCOVERABLE_HIDE_UI_IND,
+
 	/** Number of Fast Pair advertising modes. */
 	BT_FAST_PAIR_ADV_MODE_COUNT
 };
 
+/** @brief Fast Pair advertising battery mode. Used to generate advertising packet.
+ *
+ * Battery data can be included in advertising packet only if the Fast Pair Provider is in Fast Pair
+ * not discoverable advertising mode. To prevent tracking, the Fast Pair Provider should not include
+ * battery data in the advertising packet all the time.
+ */
+enum bt_fast_pair_adv_battery_mode {
+	/** Do not advertise battery data. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_NONE,
+
+	/** Show battery data UI indication. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_SHOW_UI_IND,
+
+	/** Hide battery data UI indication. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_HIDE_UI_IND,
+
+	/** Number of Fast Pair advertising battery modes. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_COUNT
+};
+
+/** @brief Index of battery value. */
+enum bt_fast_pair_battery_value_idx {
```

Comment:
Maybe we could specify it as a `bt_fast_pair_battery_component`?
e.g. `BT_FAST_PAIR_BATTERY_COMPONENT_LEFT_BUD`

Reply:
Then newly introduced Battery API would be cleaner I think

Reply:
I like your naming suggestion. The specification also uses the term components in the beginning

```When a Provider includes multiple components, it can be useful to notify the Seeker of the battery level for each component.```

I see that Zephyr uses the "_comp_" infix for Bluetooth Mesh items. I suggest we shorten the name and use ``` bt_fast_pair_battery_comp```